clang/test/CodeGen/attr-counted-by-pr88931.c

```diff
@@ -0,0 +1,22 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 4
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -O2 -Wno-missing-declarations -emit-llvm -o - %s | FileCheck %s
+
+struct foo {
+  struct bar {
+    int count;
+    int array[] __attribute__((counted_by(count)));
```

Comment:
Ideally, we should be able to compute the size here; would need to change the way we compute the "outer" type.

Can leave that for a followup, I guess.

Reply:
We can, if handling `struct bar` on its own. It's just there's seemingly no way in C to access the fields in `struct bar` from `struct foo`...

Reply:
The only reason foo is relevant in the first place is that getOuterLexicalRecordContext() skips over bar.

Reply:
I see what you mean. I'd like to leave it for a followup, since this change is meant to fix an ICE.